src/gz/bufread.rs

```diff
@@ -419,20 +568,20 @@ impl<R: BufRead> Read for GzDecoder<R> {
 
         loop {
             *inner = match mem::replace(inner, GzState::End) {
-                GzState::Header(mut buf) => {
+                GzState::Header(mut part) => {
                     let result = {
-                        let mut reader = Buffer::new(&mut buf, reader.get_mut().get_mut());
-                        read_gz_header(&mut reader)
+                        let mut reader = Buffer::new(&mut part, reader.get_mut().get_mut());
+                        read_gz_header_part(&mut reader)
```

Comment:
Why is src/gz/write.rs using `read_gz_header` ?
I guess we should have only one function instead of both `read_gz_header` and `read_gz_header_part `